### DISCUSSION TITLE
Git push constantly complains about authentication failure but does not prompt for password

### DISCUSSION THREAD

**Discussion by webquest-nz** [community/community#80843] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I have the following setup.

```bash
$ git remote -v
origin  https://github.com/webquest-nz/whanganuijobs.co.nz.git (fetch)
origin  https://github.com/webquest-nz/whanganuijobs.co.nz.git (push)
$ 
$ 
$ git config --system credential.helper  # no output
$ git config --global credential.helper  # no output
$ git config --local credential.helper   # no output
$
```

However when I try and push my latest changes to the repo I get this error:

```
$ git push
remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed for 'https://github.com/webquest-nz/whanganuijobs.co.nz.git/'
$
```

How  do I clear and reset the password for this repo?

**Comment by varunpats** [comment, marked as answer]
Hey, GitHub is no longer accepting account passwords when authenticating Git operations, you need to add a **PAT (Personal Access Token)** instead.

Here are the steps to generate personal access tokens.

[Click here](https://github.com/settings/tokens) for Token.

Step 1 - Open GitHub and log in with your credentials.

Step 2 - Click on the **Setting** menu.

Step 3 - From the Setting menu click on **Developer Settings**

Step 4 - From the Developer Settings menu, click on **Personal access token**

Step 5 - From the Personal access token, click on the **Generate new Token** button.

Step 6 - Now fill up required details like **Note, Expiration, Select scopes**. And then click on the **Generate Token** button.

Step 7 - After that, a new token has been generated. Copy that generated token and use this token to access Git with username and token.

Hope this resolves your problem. Please click on `Mark as Answer` button below, this will help me. Thanks!

**Comment by virajbhutada** [comment]
Git push constantly complains about authentication failure but does not prompt for password

I understand that you are facing authentication issues when trying to push changes to your GitHub repository. GitHub has discontinued support for password authentication, and you need to use a Personal Access Token (PAT) instead. Here's a step-by-step guide on how to generate and use a PAT:

1. Open GitHub and log in with your credentials.

2. Click on the "Settings" menu.

3. From the "Settings" menu, navigate to "Developer Settings."

4. In "Developer Settings," click on "Personal access token."

5. Click on the "Generate new Token" button.

6. Fill in the required details like "Note," "Expiration," and select scopes based on your needs.

7. Click on the "Generate Token" button.

8. A new token will be generated. Copy the generated token.

Now, you can use this token along with your GitHub username for authentication. Open your terminal and update the remote URL for your repository using the following command:

```git remote set-url origin https://username:token@github.com/webquest-nz/whanganuijobs.co.nz.git```

Replace `username` with your GitHub username and `token` with the copied PAT.

After updating the remote URL, try pushing your changes again:

```git push```

This should resolve the authentication issue. If you face any further problems, feel free to ask for assistance. Hope this helps!

**Comment by MaaeYeng2517** [comment]
not work it settion Personal Token 
modes of authentication.
fatal: Authentication failed for
